Documentation/ExtensionArchitecture/Tutorials/MinimalExtension.rst

```diff
@@ -0,0 +1,237 @@
+.. include:: /Includes.rst.txt
+.. index::
+   Extension development; Minimal extension
+   Tutorial; Minimal extension
+.. _extension-minimal:
+
+=================
+Minimal extension
+=================
+
+Learn how to create a minimal extension that contains a single PHP class.
+
+.. contents::
+   :local:
+
+Video - "Tech Tip - Minimal Extension"
+======================================
+
+The following video was produced in 2018 on TYPO3 v9. While some details have
+changed since then most parts remain the same. See the step-by-step guide below.
+
+.. youtube:: 15_HsnC_60Q
+
+Minimal extension - step-by-step
+================================
+
+.. rst-class:: bignums
+
+1.  Create a new sitepackage
+
+    Head over to the `Site Package Builder <https://www.sitepackagebuilder.com/new/>`__
+
+    Choose your desired TYPO3 version and :guilabel:`Fluid Styled Content`
+    as a base package.
+
+    Fill in the other fields. Download the site package stub.
+
+    You can find the code as it was produced at the time this tutorial was
+    written for TYPO3 v11 at
+    `speeddemo v1.0.0 <https://github.com/TYPO3-Documentation/speeddemo/releases/tag/1.0.0>`__.
+
+2.  Install the extension locally
+
+    See :ref:`Extension installation <extension-installation>`.
+
+3.  Remove all the fluff
+
+    The sitepackage builder creates many files that you don't need for this
+    simple extension.
+
+    You can delete the directories :file:`Build` and :file:`Configuration`.
+    Delete all sub folders of :file:`Classes`. Delete all files and folders
+    within :file:`Resources` except :file:`Resources/Public/Icons/Extension.svg`.
+
+    Delete all files on root level except :file:`composer.json` and
+    :file:`ext_emconf.php`. See what is left at
+    `speeddemo v1.1.0 <https://github.com/TYPO3-Documentation/speeddemo/releases/tag/1.1.0>`__.
+
+4.  The namespace and PSR-4 autoloading
```

Comment:
The sitepackage builder automatically creates this entry. Therefore if ppl follow this Tutorial the extension will also work in legacy installations. We are looking here at composer.json (as the prefered way) So that ppl can find out what the namespace is and then use it in their own PHP classes. I think adding this would be too much additional technical background for new users. Mathias also doesn't mention it in his video even though he is showing the legacy installation.

Reply:
Then we should state that as a note, so it is clear what to expect. I may assume, that some people are just copy/pasting. Mostly, because you have to remove files first for a minimum extension.

Reply:
Hmm, okay, I have a different approach, but it is okay for me.